src/components/implementation/no_interface/vkernel/vkernel.c

```diff
@@ -67,7 +67,7 @@ scheduler(void)
 					      VM_BUDGET_FIXED, VM_PRIO_FIXED, TCAP_DELEG_YIELD)) assert(0);
 		}
 
-		while (cos_sched_rcv(BOOT_CAPTBL_SELF_INITRCV_BASE, &tid, &rcving, &cycles)) ;
+		while (cos_sched_rcv(BOOT_CAPTBL_SELF_INITRCV_BASE, &tid, &rcving, &vm_pending, &cycles)) ;
```

Comment:
OK, now I'm confused why we would have both rcving and vm_pending.  Shouldn't the pending just be taken into account in the kernel when returning the `rcving` variable?  Why is this being made more complicated.

The API (as we discussed) is "set this variable to 1 if I can't effectively switch to the thread, set it to 0 if it is runnable".  So why now two variables?